packages/connector-kakao/src/index.test.ts

```diff
@@ -0,0 +1,134 @@
+import { ConnectorError, ConnectorErrorCodes } from '@logto/connector-core';
+import nock from 'nock';
+
+import createConnector, { getAccessToken } from '.';
+import { accessTokenEndpoint, authorizationEndpoint, userInfoEndpoint } from './constant';
+import { mockedConfig } from './mock';
+
+const getConfig = jest.fn().mockResolvedValue(mockedConfig);
+
+describe('kakao connector', () => {
+  describe('getAuthorizationUri', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should get a valid authorizationUri with redirectUri and state', async () => {
+      const connector = await createConnector({ getConfig });
+      const authorizationUri = await connector.getAuthorizationUri({
+        state: 'some_state',
+        redirectUri: 'http://localhost:3000/callback',
+      });
+      expect(authorizationUri).toEqual(
+        `${authorizationEndpoint}?client_id=%3Cclient-id%3E&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fcallback&response_type=code&state=some_state`
+      );
+    });
+  });
+
+  describe('getAccessToken', () => {
+    afterEach(() => {
+      nock.cleanAll();
+      jest.clearAllMocks();
+    });
+
+    it('should get an accessToken by exchanging with code', async () => {
+      nock(accessTokenEndpoint).post('').reply(200, {
+        access_token: 'access_token',
+        scope: 'scope',
+        token_type: 'token_type',
+      });
+      const { accessToken } = await getAccessToken(mockedConfig, {
+        code: 'code',
+        redirectUri: 'dummyRedirectUri',
+      });
+      expect(accessToken).toEqual('access_token');
+    });
+
+    it('throws SocialAuthCodeInvalid error if accessToken not found in response', async () => {
+      nock(accessTokenEndpoint)
+        .post('')
+        .reply(200, { access_token: '', scope: 'scope', token_type: 'token_type' });
+      await expect(
+        getAccessToken(mockedConfig, { code: 'code', redirectUri: 'dummyRedirectUri' })
+      ).rejects.toMatchError(new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+    });
+  });
+
+  describe('getUserInfo', () => {
+    beforeEach(() => {
+      nock(accessTokenEndpoint).post('').reply(200, {
+        access_token: 'access_token',
+        scope: 'scope',
+        token_type: 'token_type',
+      });
+    });
+
+    afterEach(() => {
+      nock.cleanAll();
+      jest.clearAllMocks();
+    });
+
+    it('should get valid SocialUserInfo', async () => {
+      nock(userInfoEndpoint).post('').reply(200, {
+        sub: '1234567890',
+        name: 'monalisa octocat',
+        given_name: 'monalisa',
+        family_name: 'octocat',
+        picture: 'https://github.com/images/error/octocat_happy.gif',
+        email: 'octocat@google.com',
+        email_verified: true,
+        locale: 'en',
+      });
```

Comment:
<img width="592" alt="image" src="https://user-images.githubusercontent.com/36393111/187111021-558cc9d2-96c4-4ce4-9f59-2888e5b009f2.png">
@pemassi  Looks like the mocked response structure does not match the response guard you defined here. That might cause the ut failure.